pep-0637.txt

```diff
@@ -140,6 +140,30 @@ specification would improve notation and provide additional value:
      >>> # new syntax
      >>> ds["empty"][lon=1:5, lat=6:] = 10
 
+6. Functions/methods whose arguments is another function (plus its
```

Comment:
I like this use-case, but I think it needs to be explained better. I didn't really understand it until I followed the links to the trio-run bug report and read this:

"Anyway, here's the problem: a function like run has to take the thing to run, and it also has to take its own arguments to configure how it runs it. So we need some way to distinguish which arguments are for which, but all the possible solutions involve some kind of awkwardness."

My personal feeling is that quoting the trio-run dev directly, with a link to the bug report, actual speaks stronger than the code samples. The problem with the code samples is that until you understand what they are intended to do, which requires some familiarity with trio-run, they look "good enough" -- especially the first, without partial.

If you agree, would you care to reword the rationale? If you strongly agree that the existing wording is superior, I won't argue further But I think that the final decision will be up to the PEP author.